nupic/data/record_stream.py

```diff
@@ -25,178 +25,271 @@
       '_sequenceId': the value from the sequenceId field (if any)
-    
+
+    :param inputRow: sequence of values corresponding to a single input metric
+      data row
+    :rtype: dict
     """
-    
-    values = self.getNextRecord()
-    if values is None:
-      return None
-    
-    if not values:
-      return dict()
-    
+
     # Create the return dict
-    result = dict(zip(self.getFieldNames(), values))
-    
-    # Add in the special fields
-    catIdx = self.getCategoryFieldIdx()
-    resetIdx = self.getResetFieldIdx()
-    sequenceIdx = self.getSequenceIdFieldIdx()
-    timeIdx = self.getTimestampFieldIdx()
-    learningIdx = self.getLearningFieldIdx()
+    result = dict(zip(self._fieldNames, inputRow))
 
-    if catIdx is not None:
+    # Add in the special fields
+    if self._categoryFieldIndex is not None:
       # category value can be an int or a list
-      if isinstance(values[catIdx], int):
-        result['_category'] = [values[catIdx]]
+      if isinstance(inputRow[self._categoryFieldIndex], int):
+        result['_category'] = [inputRow[self._categoryFieldIndex]]
       else:
-        result['_category'] = values[catIdx] if values[catIdx] else [None]
+        result['_category'] = (inputRow[self._categoryFieldIndex]
+                               if inputRow[self._categoryFieldIndex]
+                               else [None])
     else:
       result['_category'] = [None]
 
-    if resetIdx is not None:
-      result['_reset'] = int(bool(values[resetIdx]))
+    if self._resetFieldIndex is not None:
+      result['_reset'] = int(bool(inputRow[self._resetFieldIndex]))
     else:
       result['_reset'] = 0
-      
-    if learningIdx is not None:
-      result['_learning'] = int(bool(values[learningIdx]))
-      
+
+    if self._learningFieldIndex is not None:
+      result['_learning'] = int(bool(inputRow[self._learningFieldIndex]))
+
     result['_timestampRecordIdx'] = None
-    if timeIdx is not None:
-      result['_timestamp'] = values[timeIdx]      
+    if self._timestampFieldIndex is not None:
+      result['_timestamp'] = inputRow[self._timestampFieldIndex]
       # Compute the record index based on timestamp
       result['_timestampRecordIdx'] = self._computeTimestampRecordIdx(
-                                                        values[timeIdx])
+        inputRow[self._timestampFieldIndex])
     else:
       result['_timestamp'] = None
-    
+
     # -----------------------------------------------------------------------
     # Figure out the sequence ID
-    hasReset = resetIdx is not None
-    hasSequenceId = sequenceIdx is not None
+    hasReset = self._resetFieldIndex is not None
+    hasSequenceId = self._sequenceFieldIndex is not None
     if hasReset and not hasSequenceId:
       # Reset only
       if result['_reset']:
         try:
           self._sequenceId += 1
         except:
-          import pdb; pdb.set_trace() 
+          import pdb; pdb.set_trace()
```

Comment:
Might as well get rid of it now. No idea why we have this `import pdb; pdb.set_trace()` business here, anyway.